### PR TITLE
feat: add engine/memory_format os env variables and delay engine initialization

### PR DESCRIPTION
**Pull request by jaidisido**
### Feature or Bugfix

- Feature

### Detail
- Add engine and memory format OS environment variables
- Delay Ray engine initialization until first distributed API call

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 7d9fc3201ebe3af85dab59d409e83340b4e4e1a5
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5fa8f8da-29cb-455a-a5fc-fbeb9ad63993%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: e06bc79cbe591e8f3a32ad951aef2046f35e9488
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=79da0592-153c-4a52-99d9-fd1809869dd0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: e06bc79cbe591e8f3a32ad951aef2046f35e9488
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=61ce91ac-2c98-432c-a8fe-ddd640c37021%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: tox.ini [hunk 1]
```diff
@@ -29,6 +29,8 @@ passenv =
        AWS_SESSION_TOKEN
 setenv =
        COV_FAIL_UNDER = 74.00
+       WR_ENGINE = ray
+       WR_MEMORY_FORMAT = modin
```
Comment (LeonLuttenberger) [inline tox.ini:33]: We don't necessarily need this, right? The default with `ray` and `modin` installed would still be distributed.

File: docs/source/scale.rst [hunk 2]
```diff
@@ -16,24 +16,27 @@ Once installed, you can use the library in your code as usual:
...
     >>> print(f"Execution Engine: {wr.engine.get()}")
     >>> print(f"Memory Format: {wr.memory_format.get()}")
 
 which show that both Ray and Modin are enabled as an execution engine and memory format, respectively.
+You can switch back to non-distributed mode at any point (See `Switching modes <scale.rst#switching-modes>`__ below).
+
+Initialization of the Ray cluster is lazy and only triggered when the first distributed API is executed.
+At that point, SDK for pandas looks for an environmental variable called ``WR_ADDRESS``.
+If found, it is used to send commands to a remote cluster.
+If not found, a local Ray runtime is initialized on your machine instead.
```
[the remark this answers is not in the document]
Comment (jaidisido) [reply docs/source/scale.rst:31]: Good point, will add!

File: tox.ini (hunk 1 above)
Comment (jaidisido) [reply tox.ini:33]: Nop, not needed but it allows us to test that env variables are actually reas

File: docs/source/scale.rst [hunk 3]
```diff
@@ -133,6 +139,23 @@ This table lists the ``awswrangler`` APIs available in distributed mode (i.e. th
 |                   | ``write``                    |       ✅         |
 +-------------------+------------------------------+------------------+
 
+Switching modes
```
Comment (kukushking) [inline docs/source/scale.rst:142]: Thanks this was definitely missing from the docs

File: awswrangler/_distributed.py [hunk 4]
```diff
@@ -99,6 +103,7 @@ def dispatch_on_engine(cls, func: FunctionType) -> FunctionType:
 
         @wraps(func)
         def wrapper(*args: Any, **kw: Dict[str, Any]) -> Any:
+            cls.initialize(name=cls.get().value)
```
Comment (kukushking) [inline awswrangler/_distributed.py:106]: Can we check if `cls._initialized_engine` is set and run initialise only if it is not? I understand it's probably not an expensive operation right now but still seems a bit of a waste to run this every time we call a function

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 934fe19f86207dcd3f3c9bb88c29962288b5fd30
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=accf6679-d56a-4344-8387-3dff78348ce7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 934fe19f86207dcd3f3c9bb88c29962288b5fd30
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b7ab5f50-e21d-4a5a-ac39-01ebee71ba8f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 3fca4c839992a46a79aa7e6cdb1c8688dd383364
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9076b587-c4fb-4b2a-97b3-3704e3e1c77e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 3fca4c839992a46a79aa7e6cdb1c8688dd383364
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c9136807-97ff-4f9c-97cb-b29051ed8c9c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: awswrangler/_distributed.py (hunk 4 above)
Comment (jaidisido) [reply awswrangler/_distributed.py:106]: It's already done within the initialize ray function itself

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 24b417886a53b094936cb30bcc96cda42b281ea6
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=aa1597eb-c631-4ab1-8471-ff5f58fb4168%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 24b417886a53b094936cb30bcc96cda42b281ea6
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=26b63aca-8c85-4076-a324-66f1e2f77ff7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by LeonLuttenberger — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 0c38e418b613e27292422afccc6ccedcd4a685b7
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7adc97ed-8946-43e1-91c1-44bdcef8380f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 0c38e418b613e27292422afccc6ccedcd4a685b7
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bf385463-2f86-49ab-bc8f-37bc8aea7495%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-ATYtnXPE7MOa
* Commit ID: 0c38e418b613e27292422afccc6ccedcd4a685b7
* Result: SUCCEEDED
* [Build Logs](https://i1emq028pj.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ec55f597-b3fe-4850-9d69-6482a25bef33%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**